DICOMPlugins/DICOMTID1500Plugin.py

```diff
@@ -332,6 +360,80 @@ def enumerateDuplicateNames(self, items):
       items[idx]["name"] = item
     return items
 
+  def isConcept(self, item, coding):
+    code = item.ConceptNameCodeSequence[0]
+    return code.CodingSchemeDesignator == self.codings[coding]["scheme"] and code.CodeValue == self.codings[coding]["value"]
+
+  def loadAdditionalMeasurements(self, srUID):
+    """
+    Loads length measements as annotation rulers
+    TODO: need to generalize to other report contents
+    """
+
+    srFilePath = slicer.dicomDatabase.fileForInstance(srUID)
+    sr = dicom.read_file(srFilePath)
+
+    if not self.isConcept(sr, "imagingMeasurementReport"):
+      return sr
+
+    contents = {}
+    measurements = []
+    contents['measurements'] = measurements
+    for item in sr.ContentSequence:
+      if self.isConcept(item, "personObserver"):
+        contents['personObserver'] = item.PersonName
+      if self.isConcept(item, "imagingMeasuremnts"):
+        for contentItem in item.ContentSequence:
+          if self.isConcept(contentItem, "measurementGroup"):
+            measurement = {}
+            for measurementItem in contentItem.ContentSequence:
+              if self.isConcept(measurementItem, "trackingIdentifier"):
+                measurement['trackingIdentifier'] = measurementItem.TextValue
+              if self.isConcept(measurementItem, "trackingUniqueIdentifier"):
+                measurement['trackingUniqueIdentifier'] = measurementItem.UID
+              if self.isConcept(measurementItem, "findingSite"):
+                measurement['findingSite'] = measurementItem.ConceptCodeSequence[0].CodeMeaning
+              if self.isConcept(measurementItem, "length"):
```

Comment:
I think it is more appropriate and robust to only consider NUM type content items while looking for measurements.